staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -930,6 +930,15 @@ func (e *Store) Delete(ctx context.Context, name string, deleteValidation rest.V
 	// TODO: remove the check, because we support no-op updates now.
 	if graceful || pendingFinalizers || shouldUpdateFinalizers {
 		err, ignoreNotFound, deleteImmediately, out, lastExisting = e.updateForGracefulDeletionAndFinalizers(ctx, name, key, options, preconditions, deleteValidation, obj)
+		// Update the preconditions.ResourceVersion if set since we updated the object.
+		if preconditions.ResourceVersion != nil {
```

Comment:
we should only update this if `err == nil && deleteImmediately == true` (basically, if we're going to continue deleting the object)... also, out will likely be nil in error cases